src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -731,7 +732,8 @@ private void deleteFilesIfNecessary(ComponentLogConfiguration componentLogConfig
                 })) {
             List<Path> paths = fileStream.collect(Collectors.toList());
             long totalBytes = paths.stream().mapToLong(p -> p.toFile().length()).sum();
-            long minimumBytesToBeDeleted = totalBytes - componentLogConfiguration.getDiskSpaceLimit();
+            long minimumBytesToBeDeleted = componentLogConfiguration.getDiskSpaceLimit() == null ? 0
```

Comment:
if it is 0, I'd just return immediately, there's no work to do

Reply:
You can do that check at the top of this method, or at the call-site

Reply:
It'll return after this. I guess it's fine

    if (minimumBytesToBeDeleted <= 0 || paths.size() < 2) {
        return;
    }

Reply:
You can avoid the path listing which is expensive. Do it earlier on.

Reply:
Right. Good point